tests/e2e/specs/in-game/one-offs/5_fives.spec.js

```diff
@@ -393,19 +415,18 @@ describe('FIVES', () => {
         cy.playOneOffAndResolveAsPlayer(Card.FIVE_OF_SPADES);
         cy.window()
           .its('cuttle.gameStore')
-          .then((gameStore) => {
+          .then(async (gameStore) => {
             // Request to resolve five without discarding
-            gameStore
-              .requestResolveFive(undefined)
-              .then((res) => {
-                expect(true).to.eq(
-                  false,
-                  `Expected request to resolve five without discarding to error, but instead came back 200: ${res}`,
-                );
-              })
-              .catch((err) => {
-                expect(err).to.eq('game.snackbar.five.selectCardToDiscard');
-              });
```

Comment:
@seriouslysean and @Haviles04 I think you might get a kick out of this

This looked right but didn't actually work. Cypress would execute the `catch()` callback and show the result of the assertion, but it would pass the test before the `gameStore.requestResolveFive(undefined)` promise resolved because cypress' use of `.then()` is not a true promise. The rewrite below uses an actual promise via `async/await` so as to force cypress to await resolution/rejection of the promise from the app code before proceeding with the test.

This ensures that the test will fail if the results of the request are not what we expect. We should take a look through the test code to see if there are other examples of this pattern being used incorrectly so we can ensure that our tests of api requests like this correctly fail their respective tests if the result is not what's expected.

Reply:
This is great. It feels like an excellent time to do a write up on the problem and cement your knowledge on the topic, and potentially dive in to the nuances around it. I’d also be curious if other testing frameworks, like playwright, have this same problem.

Really nice work, @itsalaidbacklife.